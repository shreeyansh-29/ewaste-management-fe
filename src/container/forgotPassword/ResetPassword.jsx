import React, { useState } from "react";
import { Formik, Field, Form } from "formik";
import "bootstrap/dist/css/bootstrap.min.css";
import { useParams } from "react-router-dom";
import "../components/signin.css";
import { NotificationContainer } from "react-notifications";

import "./password.css";

import ShowIcon from "@mui/icons-material/VisibilityOutlined";

import ShowOffIcon from "@mui/icons-material/VisibilityOff";
import {
  CONFIRM_PASSWORD_INVALID,
  CONFIRM_PASSWORD_REQUIRED,
  PASSWORD_INVALID,
  PASSWORD_REQUIRED,
  TOAST_SUCCESS2,
} from "../constant/constant";
import api from "../../core/utilities/httpProvider";
import Toast from "../components/Toast";
function ResetPass() {
  const { token } = useParams();
  const [password, setpassword] = useState("");
  const [confirmPassword, setconfirmPsswd] = useState("");
  const [passwordErr, setErr] = useState();
  const [confirmPasswordErr, setConfirmErr] = useState();
  const [passwordType, setpasswordType] = useState("password");
  const [confirmPasswordType, setConfirmPasswordType] = useState("password");
  const [msg, setMsg] = useState("");
  const validateForm = () => {
    let formIsValid = true;

    if (!password) {
      formIsValid = false;
      setErr(PASSWORD_REQUIRED);
    } else if (!/^[a-zA-Z0-9]{6,20}$/.test(password)) {
      formIsValid = false;
      setErr(PASSWORD_INVALID);
    }
    //Confirm Password
    if (!confirmPassword) {
      formIsValid = false;
      setConfirmErr(CONFIRM_PASSWORD_REQUIRED);
    } else if (password !== confirmPassword) {
      formIsValid = false;
      setConfirmErr(CONFIRM_PASSWORD_INVALID);
    }

    return formIsValid;
  };
  const togglePassword = () => {
    if (passwordType === "password") {
      setpasswordType("text");
      return;
    }
    setpasswordType("password");
  };
  const confirmtogglePassword = () => {
    if (confirmPasswordType === "password") {
      setConfirmPasswordType("text");
      return;
    }
    setConfirmPasswordType("password");
  };
  const handleClick = async (event) => {
    event.preventDefault();
    if (validateForm()) {
      const data = {
        oldPassword: password,
        newPassword: confirmPassword,
      };

      const response = await api.post(
        `http://localhost:8083/password/save?token=${token}`,
        data
      );

      if (response.status === 200) {
        Toast.success(TOAST_SUCCESS2);
        setTimeout(() => change(), 2000);
      }
    }
  };
  const change = () => {
    setMsg("");
    window.location.href = "/Signin";
  };

  return (
    <div className="ForPassword">
      <div className="Form-body">
        <NotificationContainer />
        <Formik>
          <Form>
            <div className="psswd-heading">
              <h2
                style={{
                  textAlign: "center",
                  padding: "12px",
                  fontSize: "1.7rem",
                  fontFamily: "sans-serif",
                  color: "white",
                }}
              >
                Reset Password
              </h2>
            </div>
            <div className="Req"> {msg === "" ? "" : msg}</div>

            <div
              className="form-group"
              style={{
                marginTop: "30px",
                marginLeft: "20px",
                marginRight: "20px",
              }}
            >
              <div className="inputWithButtons">
                <Field
                  name="password"
                  className="form-control"
                  type={passwordType}
                  placeholder="New Password"
                  style={{ borderRadius: "17px" }}
                  onChange={(e) => setpassword(e.target.value)}
                  autoComplete="off"
                />
                <div className="input-group-btn">
                  <button
                    onClick={togglePassword}
                    style={{
                      border: "1px solid white",
                      backgroundColor: "white",
                    }}
                    type="button"
                  >
                    {passwordType === "password" ? (
                      <ShowOffIcon />
                    ) : (
                      <ShowIcon />
                    )}
                  </button>
                </div>
              </div>
            </div>
            <div className="formErrors1">{passwordErr}</div>
            <div
              className="form-group"
              style={{
                marginTop: "30px",
                marginLeft: "20px",
                marginRight: "20px",
              }}
            >
              <div className="inputWithButtons">
                <Field
                  name="confirm-password"
                  className="form-control"
                  type={confirmPasswordType}
                  style={{ borderRadius: "17px" }}
                  placeholder="Confirm Password"
                  onChange={(e) => setconfirmPsswd(e.target.value)}
                />
                <div className="input-group-btn">
                  <button
                    onClick={confirmtogglePassword}
                    style={{
                      border: "1px solid white",
                      backgroundColor: "white",
                    }}
                    type="button"
                  >
                    {confirmPasswordType === "password" ? (
                      <ShowOffIcon />
                    ) : (
                      <ShowIcon />
                    )}
                  </button>
                </div>
              </div>
            </div>

            <div className="formErrors1">{confirmPasswordErr}</div>

            <div className="cont" style={{ marginLeft: "25px" }}>
              <button onClick={handleClick} className="reset-button">
                Reset
              </button>
            </div>
          </Form>
        </Formik>
      </div>
    </div>
  );
}

export default ResetPass;
