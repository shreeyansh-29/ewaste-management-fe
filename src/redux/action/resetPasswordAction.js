import {
  RESET_PASSWORD_ERROR,
  // RESET_PASSWORD_REQUEST,
  RESET_PASSWORD_SUCCESS,
} from "../config/actionType";

// export const resetPasswordRequest = (data) => {
//   return {
//     type: RESET_PASSWORD_REQUEST,
//     data,
//   };
// };

export const resetPasswordSuccess = (data) => {
  return {
    type: RESET_PASSWORD_SUCCESS,
    data,
  };
};

export const resetPasswordError = (data) => {
  return {
    type: RESET_PASSWORD_ERROR,
    data,
  };
};
