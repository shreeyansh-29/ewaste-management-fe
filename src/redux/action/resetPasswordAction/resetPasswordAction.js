import * as types from "../../config/actionType";

export const resetPasswordRequest = (payload) => {
  return {
    type: types.RESET_PASSWORD_REQUEST,
    payload,
  };
};

export const resetPasswordSuccess = (payload) => {
  return {
    type: types.RESET_PASSWORD_SUCCESS,
    payload,
  };
};

export const resetPasswordError = (payload) => {
  return {
    type: types.RESET_PASSWORD_ERROR,
    payload,
  };
};
