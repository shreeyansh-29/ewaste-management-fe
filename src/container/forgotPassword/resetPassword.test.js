/* eslint-disable no-undef */
import React from "react";
import Enzyme, {mount} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import ResetPassword from "./resetPassword";
import {Form, Formik} from "formik";
import * as action from "../../redux/action/resetPasswordAction/resetPasswordAction";

Enzyme.configure({adapter: new Adapter()});

const mockedUsedNavigate = jest.fn();
const mockedUsedDispatch = jest.fn();
const mockedUsedSelector = jest.fn();
const mockStore = configureStore([]);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockedUsedDispatch,
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: () => mockedUsedSelector,
}));

describe("Reset Password", () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      resetPassword: {
        data: {
          status: 200,
        },
      },
    });
  });
  it("test resetPassword", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    );
    expect.assertions(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
describe("should test for form", () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      resetPassword: {
        data: {
          status: 200,
        },
      },
    });
  });
  it("should test for Formik and Form", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    );
    expect(wrapper.find(Formik).length).toEqual(1);
    expect(wrapper.find(Form).length).toEqual(1);
  });
  it("should have an password field", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    );
    expect(wrapper.find("#input1").at(1).length).toEqual(1);
  });
  it("should be able to enter password", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    );
    const passwordInput = wrapper.find("#input1").at(1);
    passwordInput.simulate("change", {
      persist: () => {},
      target: {
        name: "password",
        value: "123456",
      },
    });
    expect(passwordInput.html()).toMatch("123456");
  });
  it("should test togglePasswords", () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword togglePasswords={onClick} />
      </Provider>
    );
    expect(wrapper.find("#btn1").at(1).length).toEqual(1);
    wrapper.find("#btn1").at(1).simulate("click");
  });
  it("should have an confirm Password field", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    );
    expect(wrapper.find("#input2").at(1).length).toEqual(1);
    // expect(wrapper.find("#btn2").at(1).length).toEqual(1);
  });
  it("should be able to enter confirm password", () => {
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword />
      </Provider>
    );
    const cnfPasswordInput = wrapper.find("#input2").at(1);
    console.log(cnfPasswordInput);
    cnfPasswordInput.simulate("change", {
      persist: () => {},
      target: {
        name: "confirmPassword",
        value: "123456",
      },
    });
    expect(cnfPasswordInput.html()).toMatch("123456");
  });
  it("should test confirmTogglePasswords", () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword confirmTogglePasswords={onClick} />
      </Provider>
    );
    expect(wrapper.find("#btn2").at(1).length).toEqual(1);
    wrapper.find("#btn2").at(1).simulate("click");
  });
  it("should test onSubmit", () => {
    let values = {password: "123456", confirmPassword: "123456"};
    jest
      .spyOn(action, "resetPasswordRequest")
      .mockImplementation(() => Promise.resolve(values));
    const handleClick = jest.fn();

    const wrapper = mount(
      <Provider store={store}>
        <ResetPassword handleClick={handleClick} />
      </Provider>
    );
    wrapper.find(Form).simulate("submit");
    expect(handleClick).toHaveBeenCalled;
    // expect(wrapper.find('button[type="submit"]').length).toEqual(1);
    // wrapper.find(Form).simulate("submit");
    // expect(handleClick).toHaveBeenCalled;
    // console.log("wrapper", wrapper.find(Form).childAt(1).debug());
  });
});
