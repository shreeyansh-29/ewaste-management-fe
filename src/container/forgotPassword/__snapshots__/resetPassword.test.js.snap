// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password test resetPassword 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ResetPassword>
    <div
      className="ForPassword"
    >
      <div
        className="Form-body"
      >
        <Formik
          initialValues={
            Object {
              "confirmPassword": "",
              "password": "",
            }
          }
          onSubmit={[Function]}
          validationSchema={
            ObjectSchema {
              "_blacklist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_blacklistError": undefined,
              "_excludedEdges": Array [],
              "_nodes": Array [
                "confirmpassword",
                "password",
              ],
              "_sortErrors": [Function],
              "_typeError": [Function],
              "_whitelist": ReferenceSet {
                "list": Set {},
                "refs": Map {},
              },
              "_whitelistError": undefined,
              "conditions": Array [],
              "deps": Array [],
              "exclusiveTests": Object {},
              "fields": Object {
                "confirmpassword": StringSchema {
                  "_blacklist": ReferenceSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_blacklistError": undefined,
                  "_typeError": [Function],
                  "_whitelist": ReferenceSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_whitelistError": undefined,
                  "conditions": Array [],
                  "deps": Array [],
                  "exclusiveTests": Object {
                    "required": true,
                  },
                  "spec": Object {
                    "abortEarly": true,
                    "nullable": true,
                    "presence": "required",
                    "recursive": true,
                    "strict": false,
                    "strip": false,
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                  "type": "string",
                },
                "password": StringSchema {
                  "_blacklist": ReferenceSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_blacklistError": undefined,
                  "_typeError": [Function],
                  "_whitelist": ReferenceSet {
                    "list": Set {},
                    "refs": Map {},
                  },
                  "_whitelistError": undefined,
                  "conditions": Array [],
                  "deps": Array [],
                  "exclusiveTests": Object {
                    "required": true,
                  },
                  "spec": Object {
                    "abortEarly": true,
                    "nullable": true,
                    "presence": "required",
                    "recursive": true,
                    "strict": false,
                    "strip": false,
                  },
                  "tests": Array [
                    [Function],
                  ],
                  "transforms": Array [
                    [Function],
                  ],
                  "type": "string",
                },
              },
              "spec": Object {
                "abortEarly": true,
                "nullable": false,
                "presence": "optional",
                "recursive": true,
                "strict": false,
                "strip": false,
              },
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
              "type": "object",
            }
          }
        >
          <Form>
            <form
              action="#"
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <div
                className="psswd-heading"
              >
                <h2
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "sans-serif",
                      "fontSize": "1.7rem",
                      "padding": "12px",
                      "textAlign": "center",
                    }
                  }
                >
                  Reset Password
                </h2>
              </div>
              <div
                className="form-group"
                style={
                  Object {
                    "marginLeft": "20px",
                    "marginRight": "20px",
                    "marginTop": "30px",
                  }
                }
              >
                <div
                  className="inputWithButtons"
                >
                  <Field
                    autoComplete="off"
                    className="form-control"
                    id="input1"
                    name="password"
                    onChange={[Function]}
                    placeholder="New Password"
                    style={
                      Object {
                        "borderRadius": "17px",
                      }
                    }
                    type="text"
                  >
                    <input
                      autoComplete="off"
                      className="form-control"
                      id="input1"
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="New Password"
                      style={
                        Object {
                          "borderRadius": "17px",
                        }
                      }
                      type="text"
                      value=""
                    />
                  </Field>
                  <div
                    className="input-group-btn"
                  >
                    <styled.button
                      id="btn1"
                      onClick={[Function]}
                      type="button"
                    >
                      <button
                        className="sc-gsnTZi ifSsAs"
                        id="btn1"
                        onClick={[Function]}
                        type="button"
                      >
                        <ForwardRef(VisibilityOutlinedIcon)>
                          <ForwardRef(SvgIcon)
                            data-testid="VisibilityOutlinedIcon"
                          >
                            <MuiSvgIconRoot
                              aria-hidden={true}
                              as="svg"
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                              data-testid="VisibilityOutlinedIcon"
                              focusable="false"
                              ownerState={
                                Object {
                                  "children": <path
                                    d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                                  />,
                                  "color": "inherit",
                                  "component": "svg",
                                  "data-testid": "VisibilityOutlinedIcon",
                                  "fontSize": "medium",
                                  "inheritViewBox": false,
                                  "instanceFontSize": undefined,
                                  "viewBox": "0 0 24 24",
                                }
                              }
                              viewBox="0 0 24 24"
                            >
                              <Insertion
                                cache={
                                  Object {
                                    "insert": [Function],
                                    "inserted": Object {
                                      "i4bv87-MuiSvgIcon-root": true,
                                    },
                                    "key": "css",
                                    "nonce": undefined,
                                    "registered": Object {},
                                    "sheet": StyleSheet {
                                      "_alreadyInsertedOrderInsensitiveRule": true,
                                      "_insertTag": [Function],
                                      "before": null,
                                      "container": <head>
                                        <style
                                          data-styled="active"
                                          data-styled-version="5.3.5"
                                        >
                                          .ifSsAs{border:1px solid white;background-color:white;}
                                        </style>
                                        <style
                                          data-emotion="css"
                                          data-s=""
                                        >
                                          
                                          .css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}
                                        </style>
                                      </head>,
                                      "ctr": 1,
                                      "insertionPoint": undefined,
                                      "isSpeedy": false,
                                      "key": "css",
                                      "nonce": undefined,
                                      "prepend": undefined,
                                      "tags": Array [
                                        <style
                                          data-emotion="css"
                                          data-s=""
                                        >
                                          
                                          .css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}
                                        </style>,
                                      ],
                                    },
                                  }
                                }
                                isStringTag={true}
                                serialized={
                                  Object {
                                    "map": undefined,
                                    "name": "i4bv87-MuiSvgIcon-root",
                                    "next": undefined,
                                    "styles": "label:MuiSvgIcon-root;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;flex-shrink:0;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;",
                                    "toString": [Function],
                                  }
                                }
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="VisibilityOutlinedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                                />
                              </svg>
                            </MuiSvgIconRoot>
                          </ForwardRef(SvgIcon)>
                        </ForwardRef(VisibilityOutlinedIcon)>
                      </button>
                    </styled.button>
                  </div>
                </div>
              </div>
              <div
                className="form-group"
                style={
                  Object {
                    "marginLeft": "20px",
                    "marginRight": "20px",
                    "marginTop": "30px",
                  }
                }
              >
                <div
                  className="inputWithButtons"
                >
                  <Field
                    className="form-control"
                    id="input2"
                    name="confirmPassword"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    style={
                      Object {
                        "borderRadius": "17px",
                      }
                    }
                    type="text"
                  >
                    <input
                      className="form-control"
                      id="input2"
                      name="confirmPassword"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Confirm Password"
                      style={
                        Object {
                          "borderRadius": "17px",
                        }
                      }
                      type="text"
                      value=""
                    />
                  </Field>
                  <div
                    className="input-group-btn"
                  >
                    <styled.button
                      id="btn2"
                      onClick={[Function]}
                      type="button"
                    >
                      <button
                        className="sc-gsnTZi ifSsAs"
                        id="btn2"
                        onClick={[Function]}
                        type="button"
                      >
                        <ForwardRef(VisibilityOutlinedIcon)>
                          <ForwardRef(SvgIcon)
                            data-testid="VisibilityOutlinedIcon"
                          >
                            <MuiSvgIconRoot
                              aria-hidden={true}
                              as="svg"
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                              data-testid="VisibilityOutlinedIcon"
                              focusable="false"
                              ownerState={
                                Object {
                                  "children": <path
                                    d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                                  />,
                                  "color": "inherit",
                                  "component": "svg",
                                  "data-testid": "VisibilityOutlinedIcon",
                                  "fontSize": "medium",
                                  "inheritViewBox": false,
                                  "instanceFontSize": undefined,
                                  "viewBox": "0 0 24 24",
                                }
                              }
                              viewBox="0 0 24 24"
                            >
                              <Insertion
                                cache={
                                  Object {
                                    "insert": [Function],
                                    "inserted": Object {
                                      "i4bv87-MuiSvgIcon-root": true,
                                    },
                                    "key": "css",
                                    "nonce": undefined,
                                    "registered": Object {},
                                    "sheet": StyleSheet {
                                      "_alreadyInsertedOrderInsensitiveRule": true,
                                      "_insertTag": [Function],
                                      "before": null,
                                      "container": <head>
                                        <style
                                          data-styled="active"
                                          data-styled-version="5.3.5"
                                        >
                                          .ifSsAs{border:1px solid white;background-color:white;}
                                        </style>
                                        <style
                                          data-emotion="css"
                                          data-s=""
                                        >
                                          
                                          .css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}
                                        </style>
                                      </head>,
                                      "ctr": 1,
                                      "insertionPoint": undefined,
                                      "isSpeedy": false,
                                      "key": "css",
                                      "nonce": undefined,
                                      "prepend": undefined,
                                      "tags": Array [
                                        <style
                                          data-emotion="css"
                                          data-s=""
                                        >
                                          
                                          .css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}
                                        </style>,
                                      ],
                                    },
                                  }
                                }
                                isStringTag={true}
                                serialized={
                                  Object {
                                    "map": undefined,
                                    "name": "i4bv87-MuiSvgIcon-root",
                                    "next": undefined,
                                    "styles": "label:MuiSvgIcon-root;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;flex-shrink:0;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;",
                                    "toString": [Function],
                                  }
                                }
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="VisibilityOutlinedIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                                />
                              </svg>
                            </MuiSvgIconRoot>
                          </ForwardRef(SvgIcon)>
                        </ForwardRef(VisibilityOutlinedIcon)>
                      </button>
                    </styled.button>
                  </div>
                </div>
              </div>
              <div
                className="cont"
                style={
                  Object {
                    "marginLeft": "25px",
                  }
                }
              >
                <button
                  className="reset-button"
                  type="submit"
                >
                  Reset
                </button>
              </div>
            </form>
          </Form>
        </Formik>
      </div>
    </div>
  </ResetPassword>
</Provider>
`;
